docs/src/components/action/MuiLogoMenu.tsx

```diff
@@ -0,0 +1,102 @@
+
+interface MuiLogoMenuProps {
+  marginLeft: boolean;
+  smallerMargin: boolean;
+}
+
+export default function MuiLogoMenu({ smallerMargin, marginLeft }: MuiLogoMenuProps) {
+  const [contextMenu, setContextMenu] = React.useState<{
+    mouseX: number;
+    mouseY: number;
+  } | null>(null);
+
+  const handleContextMenu = (event: React.MouseEvent) => {
+    event.preventDefault();
+    setContextMenu(
+      contextMenu === null
+        ? {
+            mouseX: event.clientX + 8,
+            mouseY: event.clientY - 8,
+          }
+        : null,
+    );
+  };
+
+  const handleClose = () => {
+    setContextMenu(null);
+  };
+
+  const [copied, setCopied] = React.useState(false);
+  const handleCopy = (svgSnippet: string) => {
+    setCopied(true);
+    copy(svgSnippet).then(() => {
+      setTimeout(() => setCopied(false), 500000);
+      handleClose();
+    });
+  };
+
+  return (
+    <React.Fragment>
+      <Box
+        component={Link}
+        href="/"
+        aria-label="Go to homepage"
+        onContextMenu={handleContextMenu}
+        sx={{ mr: smallerMargin ? 1 : 1.5, ml: marginLeft ? 1.5 : undefined, cursor: 'default' }}
+      >
+        <SvgMuiLogomark height={30} width={30} />
+        <Menu
+          open={contextMenu !== null}
+          onClose={handleClose}
+          anchorReference="anchorPosition"
+          anchorPosition={
+            contextMenu !== null ? { top: contextMenu.mouseY, left: contextMenu.mouseX } : undefined
+          }
+        >
+          <MenuItem onClick={() => handleCopy(logo)}>
+            <SvgMuiLogomark
+              height={16}
+              width={18}
+              sx={(theme) => ({
+                mr: 1,
+                '> path': { fill: (theme.vars || theme).palette.text.tertiary },
+              })}
+            />
+            Copy logo as SVG
+          </MenuItem>
+          <MenuItem onClick={() => handleCopy(logoWordmark)}>
+            <TextFieldsRoundedIcon sx={{ fontSize: '18px', mr: 1, color: 'text.tertiary' }} />
+            Copy wordmark as SVG
+          </MenuItem>
+        </Menu>
+      </Box>
+      <Snackbar
```

Comment:
I'm having a weird issue when on the marketing pages... check it out:

<img width="900" alt="Screenshot 2024-05-28 at 17 13 02" src="https://github.com/mui/material-ui/assets/67129314/cacf57e0-4c28-424e-839e-ae08b995f09b">

The Snackbar has a `position: fixed` by default, and it is set here, but for some reason, that's not acting in relation to the viewport. This doesn't happen when in the docs environment. I've looked around and couldn't find an immediately obvious reason 😞 Let me know if you have any insights here.

Reply:
Found the culprit: `backdrop-filter: blur(8px);`

Here is the explanation for `filter`: https://stackoverflow.com/questions/52937708/why-does-applying-a-css-filter-on-the-parent-break-the-child-positioning

And the specification saying its the same for backdrop-filter: https://drafts.fxtf.org/filter-effects-2/#backdrop-filter-operation

Reply:
Oh wow, that's wildly unexpected! 🤯 😅 But really appreciate you for having found it & fixed it — thank you!